mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/SocialCard.kt

```diff
@@ -0,0 +1,90 @@
+package ch.epfl.sdp.mobile.ui.social
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.application.ProfileColor
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.social.SocialMode.Follow
+import ch.epfl.sdp.mobile.ui.social.SocialMode.Play
+
+/**
+ * This card is used to display player information in the Social screen, The associate button is
+ * different according to [mode] value
+ *
+ * @param person The [Person] contains the information that need to be displayed
+ * @param mode [SocialMode] define how the button visual and actions
+ * @param modifier the [Modifier] for the composable
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun SocialCard(person: Person, mode: SocialMode, modifier: Modifier = Modifier) {
```

Comment:
You may want to use a slot-based API rather than have a dedicated `mode` parameter.

---
## Slot-based API example :

```kotlin
@Composable
fun SocialCard(
  person: Person,
  trailingAction: @Composable () -> Unit,
  modifier: Modifier = Modifier,
) { ... }

// which can be later called as...
SocialCard(
  person = person,
  trailingAction = { OutlinedButton(onClick) { Text("Play")} },
)

// Notice how the slot-based API removes the need for an `onClick` callback in the `SocialCard` signature.
```

Reply:
Modified in the commit https://github.com/epfl-SDP/android/pull/91/commits/ce287617aba8794ea37f8b66b5c15d0ae9c2513a